### PR TITLE
Add benchmark retrieving a new logger

### PR DESCRIPTION
**Pull request by dmathieu**
This adds a benchmark to create a logger from a logger provider.

Related: #5054.

```
BenchmarkLoggerProviderLogger-10                         3145390               548.8 ns/op           330 B/op          1 allocs/op
```

### CODE REVIEW

**Comment by dmathieu** [comment]
New result when preallocating the logger name to avoid the use of `Sprintf` in the calculation.

```
BenchmarkLoggerProviderLogger-10                         3145390               548.8 ns/op           330 B/op          1 allocs/op
```

**Review by pellared — APPROVED** [review]

**Review by XSAM — APPROVED** [review]

File: sdk/log/provider_test.go [hunk 1]
```diff
@@ -287,3 +288,18 @@ func TestLoggerProviderForceFlush(t *testing.T) {
...
+	names := make([]string, b.N)
+	for i := 0; i < b.N; i++ {
+		names[i] = fmt.Sprintf("%d logger", i)
+	}
+
+	b.ResetTimer()
+	b.ReportAllocs()
+
+	for i := 0; i < b.N; i++ {
+		_ = p.Logger(names[i])
```
Comment (MrAlias) [inline sdk/log/provider_test.go:303]: Is the compiler optimizing this empty assignment away? Should it be stored to a var outside of the loop scope to ensure it isn't?

Comment (dmathieu) [reply sdk/log/provider_test.go:303]: Sure, I've done that. There's no difference to the results though.

**Comment by dmathieu** [comment]
New results with the logger being stored to avoid compiler optimizations.

```
BenchmarkLoggerProviderLogger-10                         2928378               542.4 ns/op           344 B/op          1 allocs/op
```

**Review by dashpole — APPROVED** [review]

**Pull request merged**